crates/core/app/src/app/mod.rs

```diff
@@ -87,7 +86,15 @@ impl App {
             .expect("state Arc should not be referenced elsewhere");
         match app_state {
             genesis::AppState::Content(app_state) => {
-                state_tx.put_chain_params(app_state.chain_params.clone());
+                // Put all the initial component parameters in the JMT:
+                // TODO: should the components themselves handle this in their
+                // `init_chain` methods?
```

Comment:
Yeah, I don't think we need to do it in this PR, but I think ideally we shouldn't have the `App` do any initialization at all, and only have it call into specific components' `init_chain` methods.  In order to do that, we'd need to have a clear assignment of responsibility of functionality per component, which we don't currently have (but should).

For instance, why is the epoch being written directly by the `App`, if the epoch handling is intertwined with the SCT? Why are the FMD parameters set by the `App` rather than by the shielded pool? etc.